lib/protocol.js

```diff
@@ -199,10 +204,51 @@ function Protocol (opts) {
 
 util.inherits(Protocol, duplexify)
 
+// extension api
+
+Protocol.prototype.remoteSupports = function (id) {
+  var self = this
+  if (!this._handshook) return
+
+  var offset = id - 64
+  var extension = self._extensions[offset]
+  if (self._available.indexOf(extension) > -1) return true
+  return false
+}
+
+Protocol.prototype.listAvailable = function () {
+  return this._available
+}
+
+Protocol.prototype.send = function (id, buf) {
+  var self = this
+  if (!self.remoteSupports(id)) return
```

Comment:
I would say that this should probably indicate if we're trying to send a message that the remote doesn't support. Otherwise, it's going to fail silently, which might be confusing.

What if this returns true or false depending on whether the remote supported the extension?